modules/metrics/mule-metrics-api/src/main/java/org/mule/runtime/metrics/api/instrument/builder/LongUpDownCounterBuilder.java

```diff
@@ -43,6 +43,15 @@ public interface LongUpDownCounterBuilder {
    */
   LongUpDownCounterBuilder withInitialValue(long initialValue);
 
+  /**
+   * Sets the meter name for the instrument to build.
+   *
+   * @param meterName the meter name.
+   *
+   * @return the {@link LongUpDownCounterBuilder}.
+   */
+  LongUpDownCounterBuilder withMeterName(String meterName);
```

Comment:
I would not set the meter name as part of the instrument builder. The meter will provide the appropriate builder associated to itself. In anycase it can internally add its name if it is useful for an instrument to know what its meter is.

Reply:
So in the implementation implementation of LongUpDownCounterBuilder that the meter provides you also has a reference to the meter to have in the instrument created.